doc/api/esm.md

```diff
@@ -1130,18 +1130,41 @@ node --experimental-top-level-await b.mjs # works
 <!-- type=misc -->
 
 To customize the default module resolution, loader hooks can optionally be
-provided via a `--experimental-loader ./loader-name.mjs` argument to Node.js.
-
+provided via `--experimental-loader ./loader-name.mjs` arguments to Node.js
 When hooks are used they only apply to ES module loading and not to any
 CommonJS modules loaded.
 
+Loaders are modules which export functions and data which can be used by
+Node.js to modify the behavior of loading ES Modules. Loaders may be "chained"
+together by passing multiple `--experimental-loader` arguments to Node.js. A
+loader module will be imported once as a singleton.
+
+A loader has access to an additional `import.meta.parentLoader` property, which
```

Comment:
Given that we already have `parentURL`, should we call this something else? E.g. if we talk about a "loader chain", `nextLoader` may be more appropriate?

Also, putting this on `import.meta` means that the loader hooks are almost completely untestable. Should this maybe go into the `context` instead?

Reply:
It is exposed on `import.meta` so access to the parent loader isn't stuck inside the loader hooks. I don't think there is anywhere else appropriate to expose it.

Reply:
I think being unable to (unit) test loader hooks is a pretty big deal. And having to access the parent loader from arbitrary top-level code seems pretty niche. Maybe for those niche cases there could be a dedicated hook (`init`) that just gets passed the `nextLoader` and the hook can choose to store it in a module-level variable if it wants to?

Making hooks harder to write and test for most use cases seems like an unfortunate trade-off here.

Reply:
@devsnek is there a reason you put this on `import.meta` instead of just having it as a function argument or accessible via `this.parentLoader` within the hooks? For example, if we renamed the last loader argument from `defaultResolve` to `parentLoader` that seems like it would satisfy Jan's request here I believe?